src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -633,14 +633,6 @@ protected function store(Request $request, Response $response)
      */
     private function restoreResponseBody(Request $request, Response $response)
     {
-        if ($request->isMethod('HEAD') || 304 === $response->getStatusCode()) {
```

Comment:
The `304 === ...` part was added a long long time ago in #1413, seemingly not covered by tests.

I think it can safely be removed because `$response` is the response taken from the cache, which cannot be a 304.